pkg/vm/vm_core_test.go

```diff
@@ -155,3 +155,212 @@ func TestOpcodeAssertionsInconsistentDst(t *testing.T) {
 		t.Error("Assertion should error out with CantWriteReturnFp")
 	}
 }
+
+func TestDeduceOp1OpcodeCall(t *testing.T) {
+	instruction := Instruction{
+		OffOp0:   1,
+		OffOp1:   2,
+		OffDst:   3,
+		DstReg:   FP,
+		Op0Reg:   AP,
+		Op1Addr:  Op1SrcAP,
+		ResLogic: ResAdd,
+		PcUpdate: PcUpdateJump,
+		ApUpdate: ApUpdateRegular,
+		FpUpdate: FpUpdateRegular,
+		Opcode:   Call,
+	}
+
+	vm := NewVirtualMachine()
+
+	_, _, err := vm.DeduceOp1(instruction, nil, nil)
+
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestDeduceOp1OpcodeAssertEqResAddWithOptionals(t *testing.T) {
+	instruction := Instruction{
+		OffOp0:   1,
+		OffOp1:   2,
+		OffDst:   3,
+		DstReg:   FP,
+		Op0Reg:   AP,
+		Op1Addr:  Op1SrcAP,
+		ResLogic: ResAdd,
+		PcUpdate: PcUpdateJump,
+		ApUpdate: ApUpdateRegular,
+		FpUpdate: FpUpdateRegular,
+		Opcode:   AssertEq,
+	}
+
+	vm := NewVirtualMachine()
+
+	dst := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(3))
+	op0 := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(2))
+
+	expected_dst := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(1))
+	expected_op0 := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(3))
+
+	m1, m2, err := vm.DeduceOp1(instruction, dst, op0)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	if *m1 != *expected_dst {
+		t.Error("Different dst value")
+	}
+	if *m2 != *expected_op0 {
+		t.Error("Different op0 value")
+	}
+}
+
+func TestDeduceOp1OpcodeAssertEqResAddWithoutOptionals(t *testing.T) {
+	instruction := Instruction{
+		OffOp0:   1,
+		OffOp1:   2,
+		OffDst:   3,
+		DstReg:   FP,
+		Op0Reg:   AP,
+		Op1Addr:  Op1SrcAP,
+		ResLogic: ResAdd,
+		PcUpdate: PcUpdateJump,
+		ApUpdate: ApUpdateRegular,
+		FpUpdate: FpUpdateRegular,
+		Opcode:   AssertEq,
+	}
+
+	vm := NewVirtualMachine()
+
+	_, _, err := vm.DeduceOp1(instruction, nil, nil)
```

Comment:
Check the returned values

Reply:
values are unchecked in rust code, should check that they are not nil?

Reply:
We should check the values